File: src/commands/list.rs
use colored::*;

use crate::vault;

/// Print all groups and key names. Values are hidden — this is a safe listing.
pub fn run() {
    let vault = vault::load();

    if vault.services.is_empty() {
        println!("{}", "  Vault is empty. Run `keypick add` to add your first group.".yellow());
        return;
    }

    println!("\n  {}\n", "Vault Contents (values hidden):".bold().underline());

    for (group, keys) in &vault.services {
        println!("  {} {}", "◆".cyan(), group.cyan().bold());
        for key in keys.keys() {
            println!("      {} {}", "·".dimmed(), key);
        }
        println!();
    }

    println!(
        "  {} {} group(s), {} key(s) total.\n",
        "→".dimmed(),
        vault.services.len(),
        vault.services.values().map(|k| k.len()).sum::<usize>()
    );
}

File: ts/src/commands/list.ts
// List all groups and key names (values hidden).
// Ported from rust/src/commands/list.rs.

import chalk from "chalk";
import * as vault from "../lib/vault.ts";

export async function run(): Promise<void> {
  const data = await vault.load();

  const groups = Object.keys(data.services).sort();
  if (groups.length === 0) {
    console.log(chalk.yellow("  Vault is empty. Run `keypick add` to add your first group."));
    return;
  }

  console.log(`\n  ${chalk.bold.underline("Vault Contents (values hidden):")}\n`);

  let totalKeys = 0;
  for (const group of groups) {
    const keys = data.services[group] ?? {};
    console.log(`  ${chalk.cyan("◆")} ${chalk.cyan.bold(group)}`);
    for (const key of Object.keys(keys).sort()) {
      console.log(`      ${chalk.dim("·")} ${key}`);
    }
    totalKeys += Object.keys(keys).length;
    console.log();
  }

  console.log(
    `  ${chalk.dim("→")} ${groups.length} group(s), ${totalKeys} key(s) total.\n`,
  );
}
